File: hello-function.c
#include <stdio.h>

int hello(char* name) {
    printf("Hello, %s!\n", name);
    return 0;
}

int main() {
    char *name = "function";            // another way to define string literals
    return hello(name);                 // Hello function!
}

/*
# This: hello-function.c
# Prev: hello-literal-string.c
# Next: fetch-function-happy.c

# Build & run:
gcc hello-function.c -o hello && ./hello
*/

File: java/HelloFunction.java
public class HelloFunction {
    private static void hello(String name) {
        // Use println is more idiomatic
        System.out.println("Hello, " + name);
    }

    public static void main(String[] args) {
        String name = "function";
        hello(name);
    }
}

/*
# This: HelloLiteralString.java
# Prev: HelloWorld.java
# Next: HelloFunction.java

# Build & run:
JNAME=HelloFunction; javac ${JNAME}.java && java ${JNAME}
*/

